core/common/src/SinkExt.kt

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+package kotlinx.io
+
+import kotlin.native.concurrent.SharedImmutable
+
+@SharedImmutable
+internal val HEX_DIGIT_BYTES = "0123456789abcdef".asUtf8ToByteArray()
+
+/**
+ * Writes two bytes containing [short], in the little-endian order, to this sink.
+ *
+ * @param short the short integer to be written.
+ *
+ * @throws IllegalStateException when the sink is closed.
+ */
+public fun <T: Sink> T.writeShortLe(short: Short) {
+    this.writeShort(short.reverseBytes())
+}
+
+/**
+ * Writes four bytes containing [int], in the little-endian order, to this sink.
+ *
+ * @param int the integer to be written.
+ *
+ * @throws IllegalStateException when the sink is closed.
+ */
+public fun <T: Sink> T.writeIntLe(int: Int) {
+    this.writeInt(int.reverseBytes())
```

Comment:
Also a note for a follow-up PR: it would be nice to reformat (but now it has to be a bit more accurate) the main sources as well to have 4 spaces as the default indent, not 2; the reasoning is as usual -- to have kotlinx consistent across the namespace and be the example